mpqp/execution/connection/aws_connection.py

```diff
@@ -1,25 +1,38 @@
 import os
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING, Any, Optional, Union
 
 from termcolor import colored
 from typeguard import typechecked
 
 if TYPE_CHECKING:
     from braket.devices.device import Device as BraketDevice
 
+from configparser import ConfigParser
+from getpass import getpass
+from pathlib import Path
+
 from mpqp.execution.connection.env_manager import get_env_variable, save_env_variable
 from mpqp.execution.devices import AWSDevice
 from mpqp.tools.errors import AWSBraketRemoteExecutionError
 
 
 def setup_aws_braket_account() -> tuple[str, list[Any]]:
-    """Setups the connection to an Amazon Braket account using user input.
+    """Set-up the connection to an Amazon Braket account using user input.
 
     This function checks whether an Amazon Braket account is already configured
-    and prompts the user to update it if needed. It then collects the user's AWS
-    access key, AWS secret key (hidden input), and the AWS region for Amazon
-    Braket. The function attempts to configure the Amazon Braket account using
-    the provided credentials.
+    and prompts the user to update it if needed. The function attempts to configure
+    the Amazon Braket account using two authentication methods:
+
+    IAM (Identity and Access Management):
+        - The user is guided to enter their AWS access key, secret access key, and region.
+        - Credentials are stored in the default AWS credentials file.
+
+    SSO (Single Sign-On):
+        - The user is guided through the process of configuring SSO authentication.
+        - SSO credentials are automatically retrieved, including the session token.
```

Comment:
This is no more true, it is not done automatically. Rewrite this part

Reply:
Modified.